From: SZALAY Attila <sasa@debian.org>
Date: Sat, 3 Oct 2015 16:21:14 +0200
Subject: I am a test patch

---
 Makefile.am | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile.am b/Makefile.am
index d78272b..b43555e 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -66,7 +66,7 @@ INSTALL_EXEC_HOOKS	=
 UNINSTALL_HOOKS		=
 CLEAN_HOOKS =
 
-AUTOMAKE_OPTIONS	= subdir-objects
+AUTOMAKE_OPTIONS	= subdir-objects foreign
 
 ${top_srcdir}/ChangeLog:
 	(cd $(top_srcdir); git log) > $@
