From: SZALAY Attila <sasa@debian.org>
Date: Wed, 20 Dec 2017 08:35:02 +0100
Subject: Remove obsoleted eventlog dependency

---
 syslog-ng.pc.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/syslog-ng.pc.in b/syslog-ng.pc.in
index 20fbdbc..c9bc892 100644
--- a/syslog-ng.pc.in
+++ b/syslog-ng.pc.in
@@ -14,6 +14,6 @@ ivykis=@with_ivykis@
 Name: syslog-ng-dev
 Description: Dev package for syslog-ng module
 Version: @PACKAGE_VERSION@
-Requires: glib-2.0 eventlog
+Requires: glib-2.0
 Libs: -L${libdir} @GLIB_LIBS@ -lsyslog-ng
 Cflags: -I${includedir}/syslog-ng @INTERNAL_IVYKIS_CFLAGS@
