From 06cdec87b9d333cbda108da3e2691e980f1eab58 Mon Sep 17 00:00:00 2001
From: Vivin Peris <you@example.com>
Date: Sun, 26 Apr 2020 22:48:58 +0530
Subject: [PATCH] crypto: fixes hang on boot due to lack of entropy

Signed-off-by: Vivin Peris  <vivinperis98@gmail.com>
---
 lib/crypto.c | 12 +++++++-----
 1 file changed, 7 insertions(+), 5 deletions(-)

diff --git a/lib/crypto.c b/lib/crypto.c
index ba714e751..d2f98ade1 100644
--- a/lib/crypto.c
+++ b/lib/crypto.c
@@ -64,7 +64,7 @@ crypto_init(void)
   openssl_init();
   openssl_crypto_init_threading();
 
-  if (RAND_status() < 0 || getenv("RANDFILE"))
+  if (getenv("RANDFILE"))
     {
       char rnd_file[256];
 
@@ -73,10 +73,12 @@ crypto_init(void)
         {
           RAND_load_file(rnd_file, -1);
           randfile_loaded = TRUE;
+          if (RAND_status() < 0)
+            {
+              fprintf(stderr,
+                      "WARNING: a trusted random number source is not available, crypto operations will probably fail. Please set the RANDFILE environment variable.");
+              g_assert_not_reached();
+            }
         }
-
-      if (RAND_status() < 0)
-        fprintf(stderr,
-                "WARNING: a trusted random number source is not available, crypto operations will probably fail. Please set the RANDFILE environment variable.");
     }
 }
